### PR TITLE
view: port wasm planner to core

### PR DESCRIPTION
**Pull request by TalDerei**
## Describe your changes

This PR adds a port of the wasm planner, integrating it within the `penumbra-view` crate.

**`SwapClaim` handling**
In the process of porting this PR, we found a bug in the handling of `SwapClaim` actions with the wasm planner.

`SwapClaim` actions produce surplus value from their prepaid fees) that must be released so that the transaction balance to zero. In principle, it should be possible to release this value into an output note. This isn't the approach we took with this PR. Instead, we release it to the transaction fee, providing a base for us to improve on later.

**Voting regression**
The original WASM planner did not handle delegator/validator voting. This PR does not yet attempt to do so either. The pcli integration test for delegator voting is commented out.

## Issue ticket number and link
References https://github.com/penumbra-zone/penumbra/issues/4081

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

> Client code

### CODE REVIEW

File: crates/view/src/planner.rs [hunk 1]
```diff
@@ -550,172 +657,92 @@ impl<R: RngCore + CryptoRng> Planner<R> {
...
+                .push(SpendPlan::new(&mut OsRng, note.note, note.position).into());
+
+            // Recompute the change outputs, without accounting for fees.
+            self.refresh_change(change_address.clone());
+            // Now re-estimate the fee of the updated transaction and adjust the change if possible.
+            let fee = self.fee_estimate(&self.gas_prices, &self.fee_tier);
+            self.adjust_change_for_fee(fee);
+
+            // Need to account to balance after applying fees.
+            self.balance = self.balance_with_fee_estimate(&self.gas_prices, &self.fee_tier);
```
Comment (erwanor) [inline crates/view/src/planner.rs:720]: ^

Comment (TalDerei) [reply crates/view/src/planner.rs:756]: can we remove this?

File: crates/view/src/planner.rs [hunk 2]
```diff
@@ -550,172 +693,116 @@ impl<R: RngCore + CryptoRng> Planner<R> {
...
+                )
+                .into());
+            };
+            self.actions
+                .push(SpendPlan::new(&mut OsRng, note.note, note.position).into());
+
+            // Recompute the change outputs, without accounting for fees.
+            self.refresh_change(change_address.clone());
+            // Now re-estimate the fee of the updated transaction and adjust the change if possible.
+            let fee = self.fee_estimate(&self.gas_prices, &self.fee_tier);
```
Comment (TalDerei) [inline crates/view/src/planner.rs:752]: shouldn't this just be `fee =`

**Comment by TalDerei** [comment]
I reviewed the surplus logic again, and checks out to me. Merge on your approval ~

**Pull request closed without merging**